### PR TITLE
DAOS-5879 dfs: Split DFS tests into unit and parallel tests

### PR DESCRIPTION
**Pull request by mchaarawi**
Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: utils/sl/fake_scons/SCons/Script/__init__.py
Comment (jolivier23) [reply utils/sl/fake_scons/SCons/Script/__init__.py:346]: I believe if you use _args, this one will go away

**Review by jolivier23 — DISMISSED** [review]

File: src/tests/ftest/daos_test/daos_core_test_dfs.yaml
Comment (saurabhtandan) [inline src/tests/ftest/daos_test/daos_core_test_dfs.yaml:18]: Minor: Why is this needed? For daos_test, nvme size was automatically created based on size of scm if I remember correctly. Is it different for dfs_test now?

**Review by saurabhtandan — DISMISSED** [review]
LGTM

File: src/tests/ftest/daos_test/daos_core_test_dfs.yaml
Comment (mchaarawi) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:18]: hmm no, i don't know where i copied that from. I will remove it when i add more tests in a future patch.

File: src/tests/ftest/util/daos_core_base.py
Comment (mchaarawi) [inline src/tests/ftest/util/daos_core_base.py:109]: @phender im looking for your feedback if this is OK (and the other change to the dfs.py file to leverage the DaosCoreBase.

File: src/tests/ftest/daos_test/daos_core_test_dfs.yaml
Comment (phender) [inline src/tests/ftest/daos_test/daos_core_test_dfs.yaml:14]: FYI, since each of the tests, `test_u` and `test_p` have their own unique avocado test timeouts and the `daos_core_test_dfs.py` does not define any other test cases, this will never be used.

**Review by phender — APPROVED** [review]
The changes look good.  The leveraging of DaosCoreBase is a good improvement.

**Comment by mchaarawi** [comment]
addressed review comments.
all failures are unrelated / ticketed.

**Review by saurabhtandan — DISMISSED** [review]
LGTM

File: utils/sl/fake_scons/SCons/Script/__init__.py
Comment (jolivier23) [inline utils/sl/fake_scons/SCons/Script/__init__.py:346]: if you set this to _args instead of args, it will shut up the warning.  Obviously there will be others in this file but at least it doesn't add a new one

File: debian/daos-tests.install
Comment (jolivier23) [inline debian/daos-tests.install:7]: I could be wrong but I would suspect we need something in the debian package version log for this change ?

**Review by jolivier23 — CHANGES_REQUESTED** [review]
check with Brian on the Debian change log requirements

File: utils/sl/fake_scons/SCons/Script/__init__.py
Comment (mchaarawi) [reply utils/sl/fake_scons/SCons/Script/__init__.py]: fixed.

File: debian/daos-tests.install
Comment (mchaarawi) [reply debian/daos-tests.install:7]: ok.. pushed update to changelog.. let's see

Comment (mchaarawi) [reply debian/daos-tests.install:7]: seems to have worked. thx

**Comment by mchaarawi** [comment]
oh dang, someone did some changes in dfs_test to add EC tests, and i need to rebase..

**Comment by mchaarawi** [comment]
this ready for re-review. I don't know why checkpatch reports failure on github. it passed on jenkins.
had to rebase a few times due to addition of DFS tests by wangdi.

**Review by jolivier23 — DISMISSED** [review]

**Comment by mchaarawi** [comment]
@saurabhtandan @phender  I would appreciate your +1 on this. I know you reviewed before but any rebase with conflict fix requires re-review.
i've had to rebase again now unfortunately with tests already passing while waiting for you to review. this touches a hot file location (daos_test.c) so it would be nice to get this landed as soon tests pass again to avoid loosing another merge race.
@jolivier23 unfortunately need another +1 from you too because of conflict.

File: src/tests/suite/SConscript
Comment (jolivier23) [inline src/tests/suite/SConscript:107]: is the indentation weird here or is that just a github artifact?

**Review by jolivier23 — APPROVED** [review]

File: src/tests/suite/SConscript
Comment (mchaarawi) [reply src/tests/suite/SConscript:107]: i think just github:
https://github.com/daos-stack/daos/blob/c9f58852b2212f53b9fdccbdfcb2dc2a365fba6d/src/tests/suite/SConscript

**Comment by saurabhtandan** [comment]
> @saurabhtandan @phender I would appreciate your +1 on this. I know you reviewed before but any rebase with conflict fix requires re-review.
> i've had to rebase again now unfortunately with tests already passing while waiting for you to review. this touches a hot file location (daos_test.c) so it would be nice to get this landed as soon tests pass again to avoid loosing another merge race.
> @jolivier23 unfortunately need another +1 from you too because of conflict.

Waiting for successful CI run.

**Comment by mchaarawi** [comment]
> > @saurabhtandan @phender I would appreciate your +1 on this. I know you reviewed before but any rebase with conflict fix requires re-review.
> > i've had to rebase again now unfortunately with tests already passing while waiting for you to review. this touches a hot file location (daos_test.c) so it would be nice to get this landed as soon tests pass again to avoid loosing another merge race.
> > @jolivier23 unfortunately need another +1 from you too because of conflict.
> 
> Waiting for successful CI run.

a green CI run is not possible like 90% of the time. the DFS tests in question have passed.
anyway, as a reviewer i think the point is to check the code changes out. Gatekeepers will verify that the CI run is OK before landing.

**Comment by saurabhtandan** [comment]
> > > @saurabhtandan @phender I would appreciate your +1 on this. I know you reviewed before but any rebase with conflict fix requires re-review.
> > > i've had to rebase again now unfortunately with tests already passing while waiting for you to review. this touches a hot file location (daos_test.c) so it would be nice to get this landed as soon tests pass again to avoid loosing another merge race.
> > > @jolivier23 unfortunately need another +1 from you too because of conflict.
> > 
> > 
> > Waiting for successful CI run.
> 
> a green CI run is not possible like 90% of the time. the DFS tests in question have passed.
> anyway, as a reviewer i think the point is to check the code changes out. Gatekeepers will verify that the CI run is OK before landing.

I meant successful CI run for the tests concerned for this PR. At the point I reviewed, The test DFS test was not run and there was a failed DFS test run for build 22.
Anyways, as a reviewer I prefer to look for successful test runs concerning the changes as well.

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by mchaarawi** [comment]
> > > > @saurabhtandan @phender I would appreciate your +1 on this. I know you reviewed before but any rebase with conflict fix requires re-review.
> > > > i've had to rebase again now unfortunately with tests already passing while waiting for you to review. this touches a hot file location (daos_test.c) so it would be nice to get this landed as soon tests pass again to avoid loosing another merge race.
> > > > @jolivier23 unfortunately need another +1 from you too because of conflict.
> > > 
> > > 
> > > Waiting for successful CI run.
> > 
> > 
> > a green CI run is not possible like 90% of the time. the DFS tests in question have passed.
> > anyway, as a reviewer i think the point is to check the code changes out. Gatekeepers will verify that the CI run is OK before landing.
> 
> I meant successful CI run for the tests concerned for this PR. At the point I reviewed, The DFS test was not run and there was a failed DFS test run for build 22.
> Anyways, as a reviewer I prefer to look for successful test runs concerning the changes as well.

anytime you see something wrong as a reviewer, please mention the issue that you have reviewed and want something addressed. when i asked for reviews, everything was running and passing :-) 
but then after some rebases caused issues.
as for build 22, i see the tests working:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3695/22/artifact/Functional/job-2020-11-11T08.15-e41aaa2-daos_test-daos_core_test_dfs/job.log
but the test later failed in shutdown. different issue it seems.

File: src/tests/suite/daos_test.c
Comment (jolivier23) [inline src/tests/suite/daos_test.c:384]: we should really put these in order...really hard to parse this

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**
